src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -373,43 +398,83 @@ public ReplicatorConfiguration([NotNull] IEndpoint target)
         public ReplicatorConfiguration([NotNull] Database database, [NotNull] IEndpoint target)
+            :this(target)
         {
-            Database = CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(database), database);
-            Target = CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(target), target);
-
-            var castTarget = Misc.TryCast<IEndpoint, IEndpointInternal>(target);
-            castTarget.Visit(this);
+            CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(database), database);
+            AddCollection(database.DefaultCollection);
         }
 
         #endregion
 
-        #region Public Methods - Scopes and Collections
+        #region Public Methods - Collections' Replicator Configurations
 
-        public void AddCollections(IList<Collection> collections, [CanBeNull]CollectionConfiguration config = null)
+        /// <summary>
+        /// Add a list of collections in the replication with an optional shared configuration. 
+        /// </summary>
+        /// <param name="collections"> that the given config will apply to</param>
+        /// <param name="config"> will apply to the given collections</param>
+        public void AddCollections(IList<Collection> collections, [CanBeNull] CollectionConfiguration config = null)
         {
+            if (collections == null || collections.Count == 0)
+                return;
+
             foreach(var col in collections) {
-                CollectionConfigs.Add(col, config);
+                AddCollection(col, config);
             }
         }
 
+        /// <summary>
+        /// Add a collection in the replication with an optional collection configuration. 
+        /// </summary>
+        /// <remarks>
+        /// The given configuration will replace the existing configuration of the given collection.
+        /// Default configuration will apply in the replication if the given configuration is null.
+        /// Configuration will be read only once applied to the collection in the replication.
+        /// </remarks>
+        /// <param name="collection"> to be added in the collections' configuration list</param>
+        /// <param name="config"> to be added in the collections' configuration list</param>
+        /// <exception cref="ArgumentNullException">will be thrown if collection is null.</exception>
         public void AddCollection(Collection collection, [CanBeNull] CollectionConfiguration config = null)
         {
-            CollectionConfigs.Add(collection, config);
+            CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(collection), collection);
+
+            if (Collections.Count > 0 && collection.Database != Database)
+                throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, $"The database {collection.Database} of the collection you are trying to add doesn't match with the {Database}.");
```

Comment:
Error message could be better:  "All collections in the configuration must operate on the same database ({Database}), but the provided collection does not (found {collection.Database})" or something